src/protobuf-net.Grpc.Reflection/FileDescriptorSetFactory.cs

```diff
@@ -34,89 +34,21 @@ private static void Populate(FileDescriptorSet fileDescriptorSet, Type serviceTy
-                };
-
-                var dependencies = new HashSet<string>();
-                foreach (var op in ContractOperation.FindOperations(binderConfiguration, serviceContract, null))
-                {
-                    // TODO: Validate op
-                    if (TryGetType(binderConfiguration, op.From, fileDescriptorSet, out var inputType, out var inputFile)
-                        && TryGetType(binderConfiguration, op.To, fileDescriptorSet, out var outputType, out var outputFile))
-                    {
-                        serviceDescriptor.Methods.Add(new MethodDescriptorProto
-                        {
-                            Name = op.Name,
-                            InputType = inputType,
-                            OutputType = outputType,
-                            ClientStreaming = op.MethodType == MethodType.ClientStreaming ||
-                                              op.MethodType == MethodType.DuplexStreaming,
-                            ServerStreaming = op.MethodType == MethodType.ServerStreaming ||
-                                              op.MethodType == MethodType.DuplexStreaming
-                        });
-
-                        dependencies.Add(inputFile);
-                        dependencies.Add(outputFile);
-                    }
-                }
-
-                var fileDescriptor = new FileDescriptorProto
-                {
-                    Name = serviceName + ".proto",
-                    Services = { serviceDescriptor },
-                    Syntax = "proto3",
-                    Package = package
-                };
-
-                foreach (var dependency in dependencies)
-                {
-                    fileDescriptor.Dependencies.Add(dependency);
-                    fileDescriptor.AddImport(dependency, true, default);
-                }
-
-                fileDescriptorSet.Files.Add(fileDescriptor);
-            }
-        }
-
-        private static bool TryGetType(BinderConfiguration binderConfiguration, Type type, FileDescriptorSet fileDescriptorSet, out string fullyQualifiedName, out string descriptorProto)
-        {
-            var typeName = type.Name;
-            var fileName = type.FullName + ".proto";
-            var fileDescriptor = fileDescriptorSet.Files.SingleOrDefault(f => f.Name.Equals(fileName, StringComparison.Ordinal));
+                BinderConfiguration = binderConfiguration,
+                ProtoSyntax = ProtoSyntax.Proto3
+            };
 
-            TypeModel model = binderConfiguration.MarshallerCache.TryGetFactory(type) is ProtoBufMarshallerFactory factory ? factory.Model : RuntimeTypeModel.Default;
-
-            if (fileDescriptor is null)
+            foreach (var serviceContract in serviceContracts)
             {
-                var schema = model.GetSchema(type, ProtoSyntax.Proto3);
-
+                if (!binderConfiguration.Binder.IsServiceContract(serviceContract, out var serviceName)) continue;
```

Comment:
Maybe we could add service name as out param to `SchemaGenerator.GetSchema()`? Then we wouldn't need to call this here. It's also called in schema generator.